morph/readability.py

```diff
@@ -0,0 +1,369 @@
+                    elif is_srt:
+                        srt_count += 1
+                        if srt_count <= 2:
+                            continue
+                        elif t == '':
+                            srt_count = 0
+                            continue
+                    filtered_text += t + '\n'
+                    if len(filtered_text) >= 2048:
+                        parse_text(filtered_text)
+                        filtered_text = ''
+
+                parse_text(filtered_text)
+
+            try:
+                with open(file_name.strip(), 'rt', encoding='utf-8') as f:
+                    input = '\n'.join([l.strip().replace(u'\ufeff', '') for l in f.readlines()])
+                    proc_lines(input, is_ass, is_srt)
+                    show = Show(file_name, seen_morphs)
+                    readability = 0.0 if i_count == 0 else 100.0 * known_count / i_count
+                    known_percent = 0.0 if len(seen_morphs.keys()) == 0 else 100.0 * len(known_morphs) / len(seen_morphs.keys())
+                    self.writeOutput('%s\t%d\t%d\t%0.2f\t%d\t%d\t%0.2f\n' % (show.name, len(seen_morphs), len(known_morphs), known_percent, i_count, known_count, readability))
+                    
+                    if save_study_plan:
+                        shows.append(show)
+            except:
+                self.writeOutput("Failed to process '%s'\n" % file_name)
+                raise
+
+        def accepted_filetype(filename):
+            lower_filename = filename.lower()
+            if lower_filename.endswith('.srt'):
+                return True
+            if lower_filename.endswith('.ass'):
+                return True
+            if lower_filename.endswith('.txt'):
+                return True
+            return False
+        
+        list_of_files = list()
+        for (dirpath, _, filenames) in os.walk(input_path):
+            list_of_files += [os.path.join(dirpath, filename) for filename in filenames if accepted_filetype(filename)]
+
+        if len(list_of_files) > 0:
+            self.writeOutput("\n[Current Readability]\n")
+            self.writeOutput('%s\t%s\t%s\t%s\t%s\t%s\t%s\n' % ("Input", "Total Morphs", "Known Morphs", "% Known Morphs", "Total Instances", "Known Instances", "% Readability"))
+
+            mw.progress.start( label='Updating data', max=len(list_of_files), immediate=True )
+            for n, file_path in enumerate(sorted(list_of_files, key=natural_keys)):
+                mw.progress.update( value=n, label='Parsing (%d/%d) %s' % (n+1, len(list_of_files), os.path.basename(file_path)) )
+                if os.path.isfile(file_path):
+                    is_ass = os.path.splitext(file_path)[1].lower() == '.ass'
+                    is_srt = os.path.splitext(file_path)[1].lower() == '.srt'
+                    measure_readability(file_path, is_ass, is_srt)
+            mw.progress.finish()
+        else:
+            self.writeOutput('\nNo files found to process.\n')
```

Comment:
If there are no files to process, is there any point in creating the study plan, word report, and frequency list?

Reply:
Done.